### DISCUSSION TITLE
Socket.io compatibility between old and new protocol.

### DISCUSSION THREAD

**Discussion by billouboq** [socketio/socket.io#4749] [Q&A]
Hello,

I would like to know if there is any way to update socket.io-client on a react-native app without breaking compatibility with old apps ?

At the moment all our apps have socket.io-client version 2 and we would like to upgrade to 4 BUT since protocol changed, I fear that we could not make communication possible between old apps user and new app users. Is there any way to make it possible ?

Thanks

**Comment by darrachequesne** [comment]
Hi! You can enable compatibility with v2 clients with the `allowEIO3` option:

```js
const io = new Server({
  allowEIO3: true
});
```

So that both v2 and v4 clients will be able to connect.

Reference: https://socket.io/docs/v4/server-options/#alloweio3
